File: sgi-frontend/src/types/core/ubigeo.ts
// Tipos para Ubigeo (Geográfico del Perú)

export interface DepartamentoGeo {
    id: number;
    nombre: string;
    ubigeo: string;
}

export interface Provincia {
    id: number;
    nombre: string;
    departamento_id: number;
    ubigeo: string;
}

export interface Distrito {
    id: number;
    nombre: string;
    provincia_id: number;
    ubigeo: string;
}

File: sgi-backend/app/schemas/core/ubigeo.py
from pydantic import BaseModel, ConfigDict

# ============================================
# Schemas para Ubigeo (Geográfico)
# ============================================

class DepartamentoGeo(BaseModel):
    """Departamento geográfico del Perú"""
    id: int
    nombre: str
    ubigeo: str
    
    model_config = ConfigDict(from_attributes=True)


class Provincia(BaseModel):
    """Provincia del Perú"""
    id: int
    nombre: str
    departamento_id: int
    ubigeo: str
    
    model_config = ConfigDict(from_attributes=True)


class Distrito(BaseModel):
    """Distrito del Perú"""
    id: int
    nombre: str
    provincia_id: int
    ubigeo: str
    
    model_config = ConfigDict(from_attributes=True)
